pkg/credentials/provider.go

```diff
@@ -76,7 +81,10 @@ func (sp *Provider) Retrieve() (credentials.Value, error) {
 	}, nil
 }
 
-func (sp *Provider) assumeRole(ctx context.Context, token string) (*sts.AssumeRoleWithWebIdentityOutput, error) {
+func (sp *Provider) assumeRole(ctx context.Context, token string) (output *sts.AssumeRoleWithWebIdentityOutput, err error) {
+	emitMetrics := sp.telemetry.Instrument([]string{"Provider", "assumeRole"}, &err)
+	defer emitMetrics()
```

Comment:
Eh, my Java-sense is tingling but I've seen objects called "cancel" returned and then "defer cancel()" so maybe this is idiomatic in Go? I feel like telemetry.Profile() returning a Profiler (or whatever) instance and then having Profiler.emitMetrics() might be more intuitive. Really what I'd like is an aspect-oriented @Profile.